### PR TITLE
Fix ipfs/go-ipfs#4224

### PR DESCRIPTION
**Pull request by jbarthelmes**
Lift (arbitrary?) restriction on file location.

Original author: @whyrusleeping

### CODE REVIEW

**Comment by jbarthelmes** [comment]
https://github.com/ipfs/go-ipfs/issues/4224#issuecomment-478550898 @Stebalien
> The only remaining attack vector is confirmation: verifying that some file exists at some path. Unfortunately, that's easily exploitable with guess and check: guess the first byte, guess the second byte, etc.

I don't understand this yet.

**Comment by Stebalien** [comment]
If an attacker could access the go-ipfs API, it could use the filestore to read an arbitrary file from the disk.

**Comment by jbarthelmes** [comment]
@Stebalien
>If an attacker could access the go-ipfs API, it could use the filestore to read an arbitrary file from the disk.

I still don't see the problem that the deleted 3 lines of code solve.

>access the go-ipfs API

Which API do you mean? I'm thinking of these scenarios:
1. Local privilege escalation: attacker gains same read access as `ipfs daemon` user. Solution: restrict that user's rights. Access to `$HOME` is already enforced by the OS.
2. Remote attacker uses HTTP API to read from file system. Solution: Fix your network security.

Either way, restricting access to `$HOME` by default does not protect sensitive files like `.ssh/id_*` or `.ipfs/config`.

**Comment by Stebalien** [comment]
Let's discuss this on the main issue. The question isn't how to lift this limitation, it's _should_ we lift it and when.

File: fsrefstore.go [hunk 1]
```diff
@@ -281,9 +281,6 @@ func (f *FileManager) putTo(b *posinfo.FilestoreNode, to putter) error {
 		if !f.AllowFiles {
 			return ErrFilestoreNotEnabled
 		}
-		if !filepath.HasPrefix(b.PosInfo.FullPath, f.root) { //nolint:staticcheck
-			return fmt.Errorf("cannot add filestore references outside ipfs root (%s)", f.root)
-		}
 
 		p, err := filepath.Rel(f.root, b.PosInfo.FullPath)
```
Comment (jbarthelmes) [inline fsrefstore.go:285]: this should be changed too, so that `dobj.FilePath` saves an absolute path

**Pull request closed without merging**